### PR TITLE
Ports Dreamseeker window flashing from TG

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Untested but probably works
This ports window flashing from TG
What the hell is with preferences though? I had to copypaste the whole thing from start to finish, why is it so much spaghetti and will anyone explain what each thing does?
The subsequent PR related to this will be having midround roles make your Dreamseeker window flash orange, preferably with options to have it off, for special/antag roles only, and for all roles.
:cl:
 * rscadd: Added a new function to the game that makes your Dreamseeker window flash orange. Currently not used by anything.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Yes I AM going to test it when I get the chance

**Comment by ShiftyRail** [comment]
Did you get an authorisation from the AGPL boogeyman

**Comment by boy2mantwicethefam** [comment]
AGPL boogeyman secured
I think it counts

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1776,3 +1776,13 @@ Game Mode config tags:
...
 	else
 		return null
+
+//Ported from TG
+/proc/window_flash(client/C, ignorepref = FALSE)
+    if(ismob(C))
+        var/mob/M = C
+        if(M.client)
+            C = M.client
+    if(!C || (!C.prefs.window_flashing && !ignorepref))
```
Comment (Intigracy) [inline code/__HELPERS/unsorted.dm]: Change `!C`to `!istype(C)`

**Review by Intigracy — CHANGES_REQUESTED** [review]
Because it might get passed a mob with no client

**Review by Intigracy — APPROVED** [review]

**Pull request merged**